clients/src/main/java/org/apache/kafka/clients/tools/VerifiableProducer.java

```diff
@@ -124,6 +132,23 @@ private static ArgumentParser argParser() {
                 .choices(0, 1, -1)
                 .metavar("ACKS")
                 .help("Acks required on each produced message. See Kafka docs on request.required.acks for details.");
+        
+        parser.addArgument("--producer-prop")
+                .action(Arguments.append())
+                .required(false)
+                .type(String.class)
+                .dest("userSpecifiedProducerProps")
+                .metavar("PRODUCER-PROP")
+                .help("Any custom producer properties of the form key=val.");
+
+        parser.addArgument("--close-timeout")
+                .action(store())
+                .required(false)
+                .setDefault(10L)
```

Comment:
My current hypothesis is that printing to stdout in each callback has enough overhead that a huge lag develops between the stuff queued up for stdout and the messages actually sent.

If you run the producer with no close timeout at with default throughput of -1, and then hit ctrl-c, you'll see many many thousands of messages printed after the fact, and wait a very long time for everything to flush. I don't think the buffer size is the issue (~16,000 by default), since many many than 16,000 messages print after hitting ctrl-c.

If you comment out the println statements in the callback and replace with some computations, the producer exits almost immediately when you hit ctrl-c.